python-package/lightgbm/basic.py

```diff
@@ -564,15 +568,15 @@ def _c_int_array(data):
     return (ptr_data, type_data, data)  # return `data` to avoid the temporary copy is freed
 
 
-def _check_for_bad_pandas_dtypes(pandas_dtypes_series):
+def is_allowed_numpy_dtype(dtype) -> bool:
```

Comment:
given that this is only used in a private function I think it should be private as well (I can't comment on the line that uses this function because it wasn't modified here but it'd need to change as well if you agree.
```suggestion
def _is_allowed_numpy_dtype(dtype) -> bool:
```

Reply:
agreed, thanks for catching that!

I did this in https://github.com/microsoft/LightGBM/pull/5704/commits/55abc4ab1d20843e20469b67b1bd3c6001dc86a8 while resolving merge conflicts, after merging #5694 .